storage/pipeline/input.go

```diff
@@ -550,12 +555,32 @@ func (m *Sealing) calcTargetExpiration(ctx context.Context, ssize abi.SectorSize
 		return 0, 0, xerrors.Errorf("getting current epoch: %w", err)
 	}
 
-	minDur, maxDur := policy.DealDurationBounds(0)
+	// if the sector isn't full, use max deal duration as the target
+	if !full {
+
+		minDur, maxDur := policy.DealDurationBounds(0)
+		minExp = ts.Height() + minDur
+
+		target = maxDur
+	}
+
+	// make sure that at least one deal in the queue is within the expiration
+	if len(candidates) > 0 && candidates[0].deal.DealProposal.EndEpoch > minExp {
+		minExp = candidates[0].deal.DealProposal.EndEpoch
+	}
+
+	// apply user minimums
+	if abi.ChainEpoch(cfg.MinUpgradeSectorExpiration)+ts.Height() > minExp {
+		minExp = abi.ChainEpoch(cfg.MinUpgradeSectorExpiration) + ts.Height()
+	}
+	if abi.ChainEpoch(cfg.MinTargetUpgradeSectorExpiration)+ts.Height() > target {
+		target = abi.ChainEpoch(cfg.MinTargetUpgradeSectorExpiration) + ts.Height()
+	}
```

Comment:
We have an issue here now. Target above is set to `MaxDur` which is a duration that is now being compared against an epoch.

Reply:
Yeah this should all be epochs